quality-debt: .agents/scripts/setup/_common.sh — PR #2027 review feedback (medium)

## Unactioned Review Feedback

**Source PR**: #2027
**File**: `.agents/scripts/setup/_common.sh`
**Reviewers**: gemini, human
**Findings**: 3
**Max severity**: medium

---

### MEDIUM: gemini (gemini-code-assist[bot])
**File**: `.agents/scripts/setup/_common.sh:110`
![medium](https://www.gstatic.com/codereviewagent/medium-priority.svg)

The command construction logic can be refactored to avoid duplicating the `extra_args` expansion pattern. This makes the code cleaner and easier to maintain if the command structure changes in the future.

```suggestion
	local cmd=()
	[[ "$use_sudo" == "true" ]] && cmd+=(sudo)
	# Use ${extra_args[@]+"${extra_args[@]}"} for safe expansion under set -u when array is empty
	cmd+=("$shell" "$tmp_script" ${extra_args[@]+"${extra_args[@]}"})
```

<details>
<summary>References</summary>

1. Use bash arrays for dynamic command construction. <sup>([link](https://github.com/marcusquinn/aidevops/blob/main/.gemini/styleguide.md))</sup>
2. In shell scripts, extract repeated logic into an internal helper function to improve maintainability. This applies even for standalone scripts where external `source` dependencies are avoided.
</details>

[View comment](https://github.com/marcusquinn/aidevops/pull/2027#discussion_r2834882517)
---

### MEDIUM: human (marcusquinn)
**File**: `.agents/scripts/setup/_common.sh:110`
Fixed in bfbe2496. Refactored to build the `cmd` array once with conditional `sudo` prepend, eliminating the duplication of the safe-expansion pattern across both branches.

[View comment](https://github.com/marcusquinn/aidevops/pull/2027#discussion_r2835024250)
---

### MEDIUM: human (marcusquinn)
**File**: `.agents/scripts/setup/_common.sh:109`
Fixed in bfbe2496. Verified the ShellCheck violations:

- **SC2034** (`REPO_URL`, `INSTALL_DIR`): These variables ARE used — by sourced modules (`setup-modules/core.sh`, `setup-modules/agent-deploy.sh`). ShellCheck can't see this because the sources use dynamic paths. Added `# shellcheck disable=SC2034` with an explanatory comment on each declaration.

- **SC1091**: The existing `# shellcheck source=` directives were not suppressing the warnings because ShellCheck can't resolve the dynamic `$SETUP_MODULES_DIR` variable. Replaced with `# shellcheck disable=SC1091` inline on each source statement (both the `$SETUP_MODULES_DIR` block and the `$(dirname ...)/setup-modules/` block).

Both files now pass `shellcheck --shell=bash` with zero violations.

[View comment](https://github.com/marcusquinn/aidevops/pull/2027#discussion_r2835024878)
---

---
_Auto-generated by `quality-feedback-helper.sh scan-merged`. Review each finding and either fix the code or dismiss with a reason._

--- a/.agents/scripts/setup/_common.sh
+++ b/.agents/scripts/setup/_common.sh
@@ -23,9 +23,13 @@ run_with_spinner() {
 	local pid
 	local spin_chars='⠋⠙⠹⠸⠼⠴⠦⠧⠇⠏'
 	local i=0
+	local output_file
+	output_file=$(mktemp)
+	# shellcheck disable=SC2064
+	trap "rm -f '$output_file'" RETURN
 
-	# Start command in background
-	"$@" &>/dev/null &
+	# Start command in background, capturing output for failure diagnosis
+	"$@" &>"$output_file" &
 	pid=$!
 
 	# Show spinner while command runs
@@ -44,7 +48,8 @@ run_with_spinner() {
 	if [[ $exit_code -eq 0 ]]; then
 		print_success "$message done"
 	else
-		print_error "$message failed"
+		print_error "$message failed. Output:"
+		cat "$output_file"
 	fi
 
 	return $exit_code
@@ -258,7 +263,9 @@ ensure_homebrew() {
 		# Add Homebrew to PATH for this session
 		local brew_prefix="/home/linuxbrew/.linuxbrew"
 		if [[ -x "$brew_prefix/bin/brew" ]]; then
-			eval "$("$brew_prefix/bin/brew" shellenv)"
+			# Use source with process substitution instead of eval (style guide: eval is forbidden)
+			# shellcheck disable=SC1090
+			source <("$brew_prefix/bin/brew" shellenv)
 		fi
 
 		# Persist to shell rc files

